haystack/nodes/file_converter/json.py

```diff
@@ -0,0 +1,102 @@
+import json
+from json import JSONDecodeError
+
+from typing import List, Optional, Any, Dict
+
+import logging
+from pathlib import Path
+
+from haystack import Document
+from haystack.nodes.file_converter import BaseConverter
+
+
+logger = logging.getLogger(__name__)
+
+
+class JsonConverter(BaseConverter):
+    """Extracts text from JSON files and casts it into Document objects."""
+
+    outgoing_edges = 1
+
+    def convert(
+        self,
+        file_path: Path,
+        meta: Optional[Dict[str, Any]] = None,
+        remove_numeric_tables: Optional[bool] = None,
+        valid_languages: Optional[List[str]] = None,
+        encoding: Optional[str] = "UTF-8",
+        id_hash_keys: Optional[List[str]] = None,
+    ) -> List[Document]:
+        """
+        Reads a JSON file and converts it into a list of Documents.
+
+        It's a wrapper around `Document.from_dict()` and, as such, acts as the inverse of `Document.to_dict()`.
+
+        It expects one of these formats:
+        - A JSON file with a list of Document dicts.
+        - A JSONL file with every line containing either a Document dict or a list of dicts.
+
+        :param file_path: Path to the JSON file you want to convert.
+        :param meta: Optional dictionary with metadata you want to attach to all resulting documents.
+                     Can be any custom keys and values.
+                     The result will have a union of metadata specified here and already present in the json.
+                     In case of same keys being used, the one passed here takes precedence/overwrites the one from the json.
+        :param remove_numeric_tables: Uses heuristics to remove numeric rows from the tables.
+                                      The tabular structures in documents might be noise for the Reader model if it
+                                      can't parse tables to find the answers. But tables
+                                      may also have long strings that could be possible candidates for searching answers.
+                                      This option retains the rows containing strings.
+        :param valid_languages: Validates languages from a list of languages specified in the [ISO 639-1]                   (https://en.wikipedia.org/wiki/ISO_639-1) format.
+                                Use this option to test for encoding errors. If the extracted text is
+                                not one of the valid languages, it may be an encoding error resulting
+                                in garbled text.
+        :param encoding: Encoding that overwrites self.encoding and is passed as the `-enc` parameter to `pdftotext`.
+                         (Run `pdftotext -listenc` in the terminal to see the list of available encodings.)
```

Comment:
I think this docstring needs to be updated since it looks like `encoding` is related to how the file is opened in python and not the `pdftotext` command line tool.